builder/parser/line_parsers.go

```diff
@@ -122,15 +125,20 @@ func parseNameVal(rest string, key string) (*Node, map[string]bool, error) {
 		if phase == inQuote {
 			if ch == quote {
 				phase = inWord
-				continue
 			}
-			if ch == '\\' {
+			// \ is special except for ' quotes - can't escape anything for '
+			if ch == '\\' && quote != '\'' {
 				if pos+1 == len(rest) {
 					phase = inWord
 					continue // just skip \ at end
 				}
+				// If next char is " then don't add the \ - its just escaping "
```

Comment:
@duglin can you please remove this along with the dead code? Even if it's useful for another PR. Also, why would we remove the backslash anyway? `echo 'hello\"world'` yields `hello\"world`.